src/sentry/api/endpoints/group_similar_issues.py

```diff
@@ -54,7 +53,7 @@ def get(self, request, group):
 
         # We need to preserve the ordering of the Redis results, as that
         # ordering is directly shown in the UI
-        for group_id, scores in zip(group_ids, group_scores):
+        for group_id, scores in list(zip(group_ids, group_scores)):
```

Comment:
Why is this necessary? Isn't it preferable to just use the iterator here rather than allocate another list?

Reply:
This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834